src/main/java/appeng/recipes/game/WirelessTerminalRecipe.java

```diff
@@ -0,0 +1,142 @@
+package appeng.recipes.game;
+
+import appeng.api.AEApi;
+import appeng.api.definitions.IComparableDefinition;
+import appeng.api.definitions.IDefinitions;
+import appeng.items.parts.ItemFacade;
+import net.minecraft.init.Items;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.InventoryCrafting;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.world.World;
+import net.minecraftforge.oredict.OreDictionary;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+
+/**
+ * Created by Tom on 10/2/2015.
+ */
+public class WirelessTerminalRecipe implements IRecipe {
+
+    private final IComparableDefinition terminal;
+    private final ItemStack waterBucket;
+
+    private static final ArrayList<String> dyes = new ArrayList<String>();
+    static {
+        dyes.add( "dyeBlack" );
+        dyes.add( "dyeBlue" );
+        dyes.add( "dyeBrown" );
+        dyes.add( "dyeCyan" );
+        dyes.add( "dyeGray" );
+        dyes.add( "dyeGreen" );
+        dyes.add( "dyeLightBlue" );
+        dyes.add( "dyeLightGray" );
+        dyes.add( "dyeLime" );
+        dyes.add( "dyeMagenta" );
+        dyes.add( "dyeOrange" );
+        dyes.add( "dyePink" );
+        dyes.add( "dyePurple" );
+        dyes.add( "dyeRed" );
+        dyes.add( "dyeWhite" );
+        dyes.add( "dyeYellow" );
+    }
+
+    public WirelessTerminalRecipe()
+    {
+        final IDefinitions definitions = AEApi.instance().definitions();
+        terminal = definitions.items().wirelessTerminal();
+        waterBucket = new ItemStack( Items.water_bucket, 1 );
+    }
+
+    @Override
+    public boolean matches( InventoryCrafting inv, World w ) {
+        return this.getOutput( inv, false ) != null;
+    }
+
+    @Nullable
+    private ItemStack getOutput( final IInventory inv, final boolean createTerminal )
```

Comment:
I'm failing to see a good way to simplify this, we sadly need to search through every possible dye to see if the second ingredient is a dye. Would it be acceptable to first check that the recipe has a Wireless Terminal, then check for a dye? This would run considerably faster, although it would still have the same complexity